src/d3a/d3a_core/simulation.py

```diff
@@ -375,13 +381,15 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
                 self.simulation_config.external_redis_communicator.\
                     publish_aggregator_commands_responses_events()
 
-                realtime_tick_length = time.time() - simulation_time_counter
+                realtime_tick_length_s = time.time() - simulation_time_counter
                 if d3a.constants.RUN_IN_REALTIME:
-                    sleep(abs(tick_lengths_s - realtime_tick_length))
-                elif self.slowdown and realtime_tick_length < tick_lengths_s:
+                    sleep(abs(tick_lengths_s - realtime_tick_length_s))
+                elif self.slot_length_realtime:
+                    sleep(abs(tick_length_realtime_s - realtime_tick_length_s))
```

Comment:
It gets me confusing with  `tick_length_realtime_s ` and `realtime_tick_length_s ` with two closely named variable and both having `realtime` in it